src/Transport/Sending/IBatcher.cs

```diff
@@ -1,10 +1,11 @@
 namespace NServiceBus.Transport.AzureServiceBus
 {
     using System.Collections.Generic;
+    using System.Threading.Tasks;
     using Transport;
 
     interface IBatcherInternal
     {
-        IList<BatchInternal> ToBatches(TransportOperations operations);
+        Task<List<BatchInternal>> ToBatches(TransportOperations operations);
```

Comment:
Why is this interface returning a Task? I would not expect any IO to happen inside a conversion function